tests/test_responses.py

```diff
@@ -309,6 +312,61 @@ async def app(scope, receive, send):
     assert response.text == "Hello, world!"
 
 
+def test_set_cookie_with_expire_datetime(test_client_factory):
+    async def app(scope, receive, send):
+        response = Response("Hello, world!", media_type="text/plain")
+        now = datetime.now(timezone.utc)
+        expires = now + timedelta(minutes=1)
+        response.set_cookie(
+            "mycookie",
+            "myvalue",
+            max_age=10,
+            expires=expires,
+            path="/",
+            domain="localhost",
+            secure=True,
+            httponly=True,
+            samesite="none",
+        )
+        await response(scope, receive, send)
+
+    client = test_client_factory(app)
+    response = client.get("/")
+    cookie: SimpleCookie = SimpleCookie(response.headers.get("set-cookie"))
+
+    assert cookie["mycookie"]
```

Comment:
What are you making sure here?

Reply:
That the cookie exists

Reply:
I personally believe that is best to keep a test to test a single thing. One test to confirm that the cookie was created. One test to check the the `expire` format.  Having a single test where we only test the format is testing both scenarios at the same time.

That said, I am probably following these rules too strictly. Nothing terrible wrong will happen, and having a single test will speed up the tests. And more than anything else, it is your project, not mine. So one test it is!